src/lib.rs

```diff
@@ -850,6 +963,259 @@
+//! # let (bob, bob_coefficients, bob_dh_sk) = Participant::new_dealer(&params, 2, &mut rng)?;
+//! # let (carol, carol_coefficients, carol_dh_sk) = Participant::new_dealer(&params, 3, &mut rng)?;
+//! #
+//! # let participants: Vec<Participant<Secp256k1Sha256>> = vec![alice.clone(), bob.clone(), carol.clone()];
+//! # let (alice_state, participant_lists) = DistributedKeyGeneration::<_, Secp256k1Sha256>::bootstrap(
+//! #       &params, &alice_dh_sk, &alice.index, &alice_coefficients, &participants, &mut rng)?;
+//! # let alice_their_encrypted_secret_shares = alice_state.their_encrypted_secret_shares()?;
+//! #
+//! # let (bob_state, participant_lists) = DistributedKeyGeneration::<_, Secp256k1Sha256>::bootstrap(
+//! #       &params, &bob_dh_sk, &bob.index, &bob_coefficients, &participants, &mut rng)?;
+//! # let mut bob_their_encrypted_secret_shares = bob_state.their_encrypted_secret_shares()?.clone();
+//! # bob_their_encrypted_secret_shares.get_mut(&alice.index).unwrap().nonce = [0; 16];
+//! #
+//! # let (carol_state, participant_lists) = DistributedKeyGeneration::<_, Secp256k1Sha256>::bootstrap(
+//! #       &params, &carol_dh_sk, &carol.index, &carol_coefficients, &participants, &mut rng)?;
+//! # let carol_their_encrypted_secret_shares = carol_state.their_encrypted_secret_shares()?;
+//! #
+//!
+//! // Bob will send an invalid share to Alice.
+//! let invalid_share = bob_their_encrypted_secret_shares.get(&alice.index).unwrap();
+//!
+//! let alice_my_encrypted_secret_shares = vec![
+//!     alice_their_encrypted_secret_shares.get(&alice.index).unwrap().clone(),
+//!     invalid_share.clone(),
+//!     carol_their_encrypted_secret_shares.get(&alice.index).unwrap().clone()];
+//!
+//! // Ignore Bob as he would be discarded anyway.
+//!
+//! let carol_my_encrypted_secret_shares = vec![
+//!     alice_their_encrypted_secret_shares.get(&carol.index).unwrap().clone(),
+//!     bob_their_encrypted_secret_shares.get(&carol.index).unwrap().clone(),
+//!     carol_their_encrypted_secret_shares.get(&carol.index).unwrap().clone()
+//! ];
+//!
+//! let (mut alice_state, alice_complaints) =
+//!     alice_state.to_round_two(alice_my_encrypted_secret_shares, &mut rng)?;
+//! let (mut carol_state, carol_complaints) =
+//!     carol_state.to_round_two(carol_my_encrypted_secret_shares, &mut rng)?;
+//!
+//! assert!(alice_complaints.len() == 1);
+//! assert!(carol_complaints.is_empty());
+//!
+//! # Ok(()) }
+//! # fn main() { assert!(do_test().is_ok()); }
+//! ```
+//!
+//! Alice shares her complaint with all other participants. Everyone will process the received complaints,
+//! update their internal state accordingly, and then conclude with this DKG. The complaint handling is done
+//! through the `blame` method which will determine whom of the plaintiff or the defendant is malicious.
+//!
+//! Note that one can process several complaints against the same participant, in which case the subsequent
```

Comment:
Hmm good point. How about having `blame` take a vector of complaints, so that internally we can get a list of pending people to remove, and process to the removal (current `remove_malicious` function) only at the end when all have been processed? That way in your example both B and C would be removed, no matter the ordering.

Reply:
Yeah, I think this would work.

Reply:
Addressed!